modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/LZ4CompressionFactory.java

```diff
@@ -0,0 +1,216 @@
+
+    /**
+     * Keep a reference to lz4 instance
+     */
+    private static final LZ4Factory factory = LZ4Factory.fastestInstance();
+
+    /**
+     * Keep a reference to lz4 fast compressor to compress data
+     */
+    private static final LZ4Compressor compressor = factory.fastCompressor();
+
+    /**
+     * Keep a reference to lz4 fast decompressor to decompress data
+     */
+    private static final LZ4FastDecompressor decompressor = factory.fastDecompressor();
+
+    /**
+     * Compress the content of the message, using andes message part list
+     *
+     * @param partList              Original content chunk list
+     * @param originalContentLength Total content length of the original message content
+     * @return Compressed message content as an AndesMessagePart
+     */
+    public AndesMessagePart getCompressedMessage(List<AndesMessagePart> partList, int originalContentLength) {
+
+        byte[] messageData = getByteArrayFromPartList(partList, originalContentLength);
+
+        // Compress message content
+        int maxCompressedLength = compressor.maxCompressedLength(originalContentLength);
+        byte[] compressed = new byte[maxCompressedLength];
+        int compressedLength =
+                compressor.compress(messageData, 0, originalContentLength, compressed, 0, maxCompressedLength);
+
+        // Final compressed array
+        byte[] compressedMessage = Arrays.copyOf(compressed, compressedLength);
+
+        // Getting andes message part of the compressed message content and returning
+        return getAndesMessagePart(compressedMessage, partList.get(0).getMessageID());
+    }
+
+    /**
+     * Decompress the compressed message content, before send to subscriber
+     *
+     * @param messagePartList       Compressed message content as a collection of andes message parts
+     * @param originalContentLength Total content length of the original message content
+     * @param maxChunkSize          Maximum allowed chunk size to be stored in DB
+     * @param messageID             Message ID of the message
+     * @return Decompressed message content as a map of andes message parts and offsets
+     */
+    public Map<Integer, AndesMessagePart> getDecompressedMessage(Collection<AndesMessagePart> messagePartList, int
+            originalContentLength, int maxChunkSize, long messageID) {
+        byte[] compressedMessageContent = getByteArrayFromPartList(messagePartList, originalContentLength);
+
+        // Decompress message content
+        byte[] decompressedMessage = new byte[originalContentLength];
+        decompressor.decompress(compressedMessageContent, 0, decompressedMessage, 0, originalContentLength);
+
+        // Creating message part map using the decompressed message.
+        // Size of this map is equals to the minimum number of andes message parts that are needed
+        Map<Integer, AndesMessagePart> messagePartMapToDeliver = new HashMap<>((int) (Math.ceil((decompressedMessage
```

Comment:
Any possibility to improve readability ? maybe doing the arithmetic operations separately and giving the final value to the HashMap constructor ?